picky-signtool/src/main.rs

```diff
@@ -0,0 +1,212 @@
+const ARG_LOGGING_INFO: &str = "info";
+const ARG_LOGGING_WARN: &str = "warn";
+const ARG_LOGGING_ERR: &str = "err";
+const ARG_LOGGING_CRITICAL: &str = "critical";
+
+fn main() -> anyhow::Result<()> {
+    let matches = config();
+
+    match matches.value_of(ARG_LOGGING) {
+        Some(log_level) => {
+            let log_level = match log_level {
+                ARG_LOGGING_TRACE => LogLevel::LogTrace,
+                ARG_LOGGING_DEBUG => LogLevel::LogDebug,
+                ARG_LOGGING_INFO => LogLevel::LogInfo,
+                ARG_LOGGING_WARN => LogLevel::LogWarn,
+                ARG_LOGGING_ERR => LogLevel::LogErr,
+                ARG_LOGGING_CRITICAL => LogLevel::LogCritical,
+                _ => panic!("No more values of log_level exists"),
+            };
+            lief::enable_logging(log_level);
+        }
+
+        None => lief::disable_logging(),
+    }
+
+    let binary_path = matches
+        .value_of(ARG_INPUT)
+        .expect("Path to a Windows executable is required");
+
+    let binary_path = PathBuf::from(binary_path);
+    let binary_name = binary_path
+        .as_path()
+        .file_name()
+        .map(|name| name.to_str())
+        .flatten()
+        .map(|name| name.to_owned())
+        .expect("file name should be present");
+
+    let binary = Binary::new(binary_path).map_err(|err| anyhow!("Failed to load the executable: {}", err))?;
+
+    if let Some(verification_checks) = matches.value_of(ARG_VERIFY) {
+        let check_flag = match verification_checks {
+            ARG_VERIFY_DEFAULT => VerificationChecks::DEFAULT,
+            ARG_VERIFY_HASH_ONLY => VerificationChecks::HASH_ONLY,
+            ARG_VERIFY_LIFETIME_SIGNING => VerificationChecks::LIFETIME_SIGNING,
+            ARG_VERIFY_SKIP_CERT_TIME => VerificationChecks::SKIP_CERT_TIME,
+            _ => panic!("No more values of `verify` parameters exists"),
+        };
+
+        let check_result = binary
+            .check_signature(check_flag)
+            .map_err(|err| anyhow!("Failed to check signature: {}", err))?;
+
+        println!("Verify the executable signature result: {:?}", check_result);
+    }
+
+    if matches.is_present(ARG_SIGN) {
+        if let (Some(certfile), Some(private_key), Some(output_path)) = (
+            matches.value_of(ARG_CERTFILE),
+            matches.value_of(ARG_PRIVATE_KEY),
+            matches.value_of(ARG_OUTPUT),
+        ) {
+            let certfile = read_file_into_vec(PathBuf::from(certfile))?;
+            let private_key = read_file_into_vec(PathBuf::from(private_key))?;
+
+            binary
+                .set_authenticode(certfile, private_key, Some(binary_name))
+                .map_err(|err| anyhow!("Failed to check signature: {}", err))?;
+
+            binary
+                .build(PathBuf::from(output_path), false)
+                .map_err(|err| anyhow!("Failed to build the signed executable: {}", err))?;
+
+            println!("Signed binary_name successfully!");
```

Comment:
binary_name ?

Reply:
Shoud be `println!("Signed {} successfully!", binary_name);`